docs: solidity recomended workflow note (#2003)
succinctlabs/sp1: book/docs/verification/onchain/solidity-sdk.md

```

> The recommended workflow is to use the `ISP1Verifier` interface on the [`SP1VerifierGateway`](./contract-addresses.md),

Reviewer: ```suggestion
When deploying applications on-chain, we recommended using the SP1 verifiers with the `ISP1Verifier` interface on the address of the [`SP1VerifierGateway`](./contract-addresses.md),
```

> The recommended workflow is to use the `ISP1Verifier` interface on the [`SP1VerifierGateway`](./contract-addresses.md),

Reviewer: ```suggestion
The recommended on-chain SP1 proof verification workflow is to use the `ISP1Verifier` interface on the [`SP1VerifierGateway`](./contract-addresses.md),
```

> and the `SP1VerifierGateway` will automatically route your proof to the correct verifier.

Reviewer: ```suggestion
so the `SP1VerifierGateway` automatically routes your proof to the correct verifier.
```

> It’s also recommended to have your program’s verification key be upgradeable within your contract, in case you want to switch to a different version of SP1.

Reviewer: ```suggestion
Your program’s verification key should be upgradeable within your contract, in case you want to upgrade to a newer version of SP1 or modify your program.
```

> Succinct also maintains the ability to freeze verifiers in the event of a security issue,

Reviewer: ```suggestion
Succinct maintains the ability to freeze verifiers on the canonical verifier gateway in the event of a security issue to prevent abuse, but verifier contract deployment is permissionless to enable customizable security configurations.
```

> but do note, anyone is free to deploy their own verifier contract. 

Reviewer: ```suggestion
```
